mad-dog/src/helpers.js

```diff
@@ -313,16 +319,94 @@ const r6 = (withNum = false) => genRandomString(6, withNum)
 /** Delay execution for n ms */
 const delay = ms => new Promise(resolve => setTimeout(resolve, ms))
 
-/** Wrapper for custom delay time */
-const waitForIndexing = async (waitTime = 5000) => {
-  logger.info(`Pausing ${waitTime}ms for discprov indexing...`)
-  await delay(waitTime)
+const getLatestIndexedBlock = async (endpoint = DISCOVERY_NODE_ENDPOINT) => {
+  return (await axios({
+    method: 'get',
+    baseURL: endpoint,
+    url: '/health_check'
+  })).data.latest_indexed_block
+}
+
+const getLatestIndexedIpldBlock = async (endpoint = DISCOVERY_NODE_ENDPOINT) => {
+  return (await axios({
+    method: 'get',
+    baseURL: endpoint,
+    url: '/ipld_block_check'
+  })).data.data.block
 }
 
-const waitForSync = async (waitTime = 20000) => {
-  logger.info(`Pausing ${waitTime}ms for sync to occur...`)
-  await delay(waitTime)
+const monitorAllUsersSyncStatus = async ({ i, libs, executeOne }) => {
```

Comment:
nit: this new name leaves me pretty unclear as to what this fn does (what does "monitor" mean? does it just log stuff out? when does it return?) - I think something like the original that implies that this waits until all of the users secondaries are caught up would be helpful